src/xcode/ENA/ENA/Source/AppDelegate & Globals/__tests__/ENAUITestsQuickActions.swift

```diff
@@ -0,0 +1,147 @@
+		XCTAssertTrue(appIcon.waitForExistence(timeout: .short))
+		if !appIcon.isHittable {
+			springboard.swipeLeft()
+		}
+		XCTAssertTrue(appIcon.isHittable)
+		appIcon.press(forDuration: 1.5)
+
+		let actionButton = springboard.buttons[newDiaryEntryLabel]
+		XCTAssertFalse(actionButton.exists, "Shortcuts should not be available on 'fresh' installations which aren't onboarded")
+    }
+
+	func testLaunchAfterOnboarding() throws {
+		let app = XCUIApplication()
+		app.setDefaults()
+		app.launchArguments.append(contentsOf: ["-isOnboarded", "YES"])
+		app.launchArguments.append(contentsOf: ["-setCurrentOnboardingVersion", "YES"])
+		app.launch()
+
+		// On home screen?
+		XCTAssertTrue(app.buttons[AccessibilityIdentifiers.Home.submitCardButton].waitForExistence(timeout: .medium))
+		// back out
+		XCUIDevice.shared.press(.home)
+
+		// check app menu
+		let appIcon = try XCTUnwrap(springboard.icons[cwaBundleDisplayName])
+		XCTAssertTrue(appIcon.waitForExistence(timeout: .short))
+		if !appIcon.isHittable {
+			springboard.swipeLeft()
+		}
+		XCTAssertTrue(appIcon.isHittable)
+		appIcon.press(forDuration: 1.5)
+
+		let actionButton = springboard.buttons[newDiaryEntryLabel]
+		XCTAssertTrue(actionButton.waitForExistence(timeout: .short))
+		actionButton.tap()
+		XCTAssertTrue(app.segmentedControls[AccessibilityIdentifiers.ContactDiary.segmentedControl].waitForExistence(timeout: .short))
+	}
+
+	func testShortcutAvailabilityDuringSubmissionFlow() throws {
+		let app = XCUIApplication()
+		app.setDefaults()
+		app.launchArguments.append(contentsOf: ["-isOnboarded", "YES"])
+		app.launchArguments.append(contentsOf: ["-setCurrentOnboardingVersion", "YES"])
+		app.launchArguments.append(contentsOf: ["-ENStatus", ENStatus.active.stringValue])
+		app.launchArguments += [UITestingParameters.ExposureSubmission.useMock.rawValue]
+		app.launchArguments.append(contentsOf: ["-testResult", TestResult.positive.stringValue])
+		app.launch()
+
+		// Open Intro screen.
+		XCTAssertTrue(app.cells.buttons[AccessibilityIdentifiers.Home.submitCardButton].waitForExistence(timeout: .long))
+		app.cells.buttons[AccessibilityIdentifiers.Home.submitCardButton].tap()
+
+		// in submission flow, that's ok
+		XCTAssertTrue(app.buttons["AppStrings.ExposureSubmission.primaryButton"].waitForExistence(timeout: .short))
+		// back out
+		XCUIDevice.shared.press(.home)
+
+		// check app menu
+		let appIcon = try XCTUnwrap(springboard.icons[cwaBundleDisplayName])
+		XCTAssertTrue(appIcon.waitForExistence(timeout: .short))
+		if !appIcon.isHittable {
+			springboard.swipeLeft()
+		}
+		XCTAssertTrue(appIcon.isHittable)
+		appIcon.press(forDuration: 1.5)
+
+		let actionButton = springboard.buttons[newDiaryEntryLabel]
+		XCTAssertFalse(actionButton.exists, "Shortcuts should not be available once the user is in submission flow")
+	}
+
+	// MARK: - Install/Uninstall our app
+
+	/// Uninstalling the app manually, if present.
+	///
+	/// No hacks and `.resolve()` involved!
```

Comment:
Dont understand this comment.

Reply:
I'm just not copying the SO stuff you find if you look up for this use case. Just leave it as it is.

Reply:
Could you please change it.

Because, I still don't get it. What is the context of this comment? What do you want to communicate to other developers? I think this comment confuses me more than it helps me. This is going on in my head if I read it: "No hacks" – ok, good code without a hack. As I would expect; "`.resolve()` involved" – what is .resolve() and why is it mentioned here? Do I need to care about this here? And if yes, why do I need to care about this?